test/e2e/scalenodes.go

```diff
@@ -75,11 +81,41 @@ var _ = Describe("Scale nodes", func() {
 		err = waitForScale(mss.Items[0].Name)
 		Expect(err).NotTo(HaveOccurred())
 	})
+
+	Specify("operator should maintain at least three worker replicas", func() {
+		ctx := context.Background()
+		infraId, err := clients.AROClusters.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		mss, err := clients.MachineAPI.MachineV1beta1().MachineSets(machineSetsNamespace).List(ctx, metav1.ListOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(mss.Items).NotTo(BeEmpty())
+
+		// Remove a replica from a worker MachineSet, and wait
+		err = scale(mss.Items[0].Name, -1)
+		Expect(err).NotTo(HaveOccurred())
+
+		ms, err := clients.MachineAPI.MachineV1beta1().MachineSets(machineSetsNamespace).Get(ctx, mss.Items[0].Name, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		err = waitForScale(mss.Items[0].Name)
+		Expect(err).NotTo(HaveOccurred())
+
+		matches, err := regexp.Match(infraId.Spec.InfraID, []byte(mss.Items[0].Name))
+		Expect(err).NotTo(HaveOccurred())
+
+		// Expect MachineSet controller to have blocked the change, unless custom MachineSet
```

Comment:
Controller is not a webhook - It does not block: it reconciles after the fact of the change.